Randomizer.SMZ3/World.cs

```diff
@@ -97,12 +101,21 @@ class World {
         }
 
         public bool CanAquireAll(Progression items, params RewardType[] rewards) {
-            return Regions.OfType<IReward>().Where(x => rewards.Contains(x.Reward)).All(x => x.CanComplete(items));
+            return rewardLookup[rewards.Sum(x => (int)x)].All(x => x.CanComplete(items));
+        }
+
+        public bool CanAquireAllMask(Progression items, int bitMask) {
```

Comment:
I see no reason why `CanAcquireAll` can't be overloaded (including the opportunity to fix the spelling), especially if the argument is typed as `RewardType`. In fact, with `RewardType` no call site need to use the `params` method.